esmvalcore/cmor/_fixes/cordex/cordex_fixes.py

```diff
@@ -0,0 +1,168 @@
+
+
+class MOHCHadREM3GA705(Fix):
+    """General fix for MOHC-HadREM3-GA7-05."""
+
+    def fix_metadata(self, cubes):
+        """Fix time long_name, and latitude and longitude var_name.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            cube.coord('latitude').var_name = 'lat'
+            cube.coord('longitude').var_name = 'lon'
+            cube.coord('time').long_name = 'time'
+
+        return cubes
+
+
+class TimeLongName(Fix):
+    """Fixes for time coordinate."""
+
+    def fix_metadata(self, cubes):
+        """
+        Fix time long_name.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            cube.coord('time').long_name = 'time'
+
+        return cubes
+
+
+class CLMcomCCLM4817(Fix):
+    """Fixes for CLMcom-CCLM4-8-17."""
+
+    def fix_metadata(self, cubes):
+        """Fix calendars.
+
+        Set calendar to 'proleptic_gregorian' to avoid
+        concatenation issues between historical and
+        scenario runs.
+
+        Fix dtype value of coordinates and coordinate bounds.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+
+        """
+        for cube in cubes:
+            if cube.coord('time').units.calendar == 'standard':
+                cube.coord('time').units = Unit(
+                    'days since 1850-1-1 00:00:00',
+                    calendar='proleptic_gregorian'
+                )
+            for coord in cube.coords():
+                if coord.dtype in ['>f8', '>f4']:
+                    coord.points = coord.core_points().astype(
+                        np.float64, casting='same_kind')
+                    if coord.bounds is not None:
+                        coord.bounds = coord.core_bounds().astype(
+                            np.float64, casting='same_kind')
+        return cubes
+
+
+class AllVars(Fix):
+    """General CORDEX grid fix."""
+
+    def _fix_rotated_coords(self, cube):
+        """Fix rotated coordinates."""
+        data_domain = self.extra_facets['domain']
+        domain = cx.cordex_domain(data_domain, add_vertices=True)
+        domain_info = cx.domain_info(data_domain)
+        for dim_coord in ['rlat', 'rlon']:
+            old_coord = cube.coord(domain[dim_coord].standard_name)
+            old_coord_dims = old_coord.cube_dims(cube)
+            points = domain[dim_coord].data
+            coord_system = iris.coord_systems.RotatedGeogCS(
+                grid_north_pole_latitude=domain_info['pollat'],
+                grid_north_pole_longitude=domain_info['pollon']
+            )
+            new_coord = iris.coords.DimCoord(
```

Comment:
It would be good to log the differences between the original grid and the pycordex one. Maybe we could raise an error for the time being when the differences are larger than a threshold of 10e-4 deg (~11km). The same for the geographical coords. We could later discuss how to tackle the grids that do not conform to the standards in an other pull request

Reply:
looks great to me